go-controller/pkg/metrics/master.go

```diff
@@ -568,3 +613,70 @@ func getPodUIDFromPortBinding(row *sbdb.PortBinding) kapimtypes.UID {
 	}
 	return kapimtypes.UID(podUID)
 }
+
+func incrementNbCfg(nbClient libovsdbclient.Client) {
+	nbCfgValue += 1
+	if err := libovsdbops.UpdateNBGlobalNbCfg(nbClient, nbCfgValue); err != nil {
+		klog.Errorf("Failed to increment NB_Global nb_cfg value: %v", err)
+		return
+	}
+	metricNbcfgNbdb.Set(float64(nbCfgValue))
+}
+
+// chassisPrivateScraper implements the Collector interface so we can
+// generate chassis metrics on-the-fly per scrape
+type chassisPrivateScraper struct {
+	sbClient libovsdbclient.Client
+}
+
+func (c *chassisPrivateScraper) Describe(ch chan<- *prometheus.Desc) {
```

Comment:
@martinkennelly I don't see this being called anywhere? Is it internally called every 30 seconds by Prometheus?

Reply:
This is called by the Prometheus client library when the program registers the collector.

Reply:
In practice it's called only once when the program starts.

Reply:
Ah ok thank you! @simonpasquier